### PR TITLE
RPC publish optional work_watch flag to add to work_watcher

### PR DESCRIPTION
**Pull request by argakiig**
Adds flag work_watch, default true, to rpc publish allowing for blocks processed via RPC call to not be eligible for dropping from active ensuring the block is not dropped from active_transactions until confirmed. Messages will continue to originate from the block as long as the block is not confirmed

With this change publishes originating from the local node will not be dropped ensuring your blocks are always broadcast and can propagate the network

### CODE REVIEW

File: nano/node/network.cpp [hunk 1]
```diff
@@ -215,15 +215,21 @@ bool nano::network::send_votes_cache (std::shared_ptr<nano::transport::channel>
 	return result;
 }
 
-void nano::network::flood_message (nano::message const & message_a)
+void nano::network::flood_message (nano::message const & message_a, bool const & is_dropable)
```
Comment (wezrule) [inline nano/node/network.cpp]: Minor typo `is_droppable`. Passing a bool by ref won't be faster, primitives are generally passed by value. Also it should probably have the `_a` suffix to be consistent with the existing parameter. Same in `flood_block`

File: nano/node/network.cpp [hunk 2]
```diff
@@ -215,15 +215,21 @@ bool nano::network::send_votes_cache (std::shared_ptr<nano::transport::channel>
...
 {
 	auto list (list_fanout ());
 	for (auto i (list.begin ()), n (list.end ()); i != n; ++i)
 	{
-		(*i)->send (message_a);
+		(*i)->send (message_a, nullptr, is_dropable);
 	}
 }
 
+void nano::network::flood_block (std::shared_ptr<nano::block> block_a, bool const & is_dropable)
```
Comment (SergiySW) [inline nano/node/network.cpp]: Probably still can be in network.hpp

Comment (argakiig) [reply nano/node/network.cpp]: was moved when I added something to the function, good catch, moved it back

File: nano/node/network.cpp (hunk 1 above)
Comment (argakiig) [reply nano/node/network.cpp]: thanks :)

File: nano/node/blockprocessor.cpp [hunk 3]
```diff
@@ -382,7 +387,7 @@ void nano::block_processor::process_live (nano::block_hash const & hash_a, std::
 	});
 }
 
-nano::process_return nano::block_processor::process_one (nano::transaction const & transaction_a, nano::unchecked_info info_a)
+nano::process_return nano::block_processor::process_one (nano::transaction const & transaction_a, nano::unchecked_info info_a, const bool work_watcher_a)
```
Comment (guilhermelawless) [inline nano/node/blockprocessor.cpp]: Shouldn't it also be called `watch_work_a` here?

Comment (argakiig) [reply nano/node/blockprocessor.cpp]: yup, thank you I thought I got them all but good catch

File: nano/node/json_handler.cpp [hunk 4]
```diff
@@ -2818,6 +2818,7 @@ void nano::json_handler::payment_wait ()
 void nano::json_handler::process ()
 {
 	const bool json_block_l = request.get<bool> ("json_block", false);
+	const bool watch_work_l = request.get<bool> ("watch_work", true);
```
Comment (SergiySW) [inline nano/node/json_handler.cpp]: Probably this option requires protection with enable_control? Same as work_generate

Comment (argakiig) [reply nano/node/json_handler.cpp]: handled same way as process force

**Pull request merged**